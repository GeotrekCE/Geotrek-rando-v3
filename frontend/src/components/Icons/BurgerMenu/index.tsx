import React from 'react';
import { GenericIconProps } from '../types';

export const BurgerMenu: React.FC<GenericIconProps> = ({
  color = 'currentColor',
  opacity,
  className,
  size,
}) => {
  return (
    <svg
      width={size}
      height={size}
      viewBox="0 0 24 25"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      className={className}
      opacity={opacity}
    >
      <path
        d="M3 12h18M3 6h18M3 18h18"
        stroke={color}
        strokeWidth={2}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};
