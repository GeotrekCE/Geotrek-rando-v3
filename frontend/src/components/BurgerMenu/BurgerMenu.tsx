import BurgerMenuSection from 'components/BurgerMenuSection/BurgerMenuSection';
import { slide as Slide } from 'react-burger-menu';
import { BurgerMenu as BmIcon } from '../Icons/BurgerMenu';
import { Cross } from '../Icons/Cross';

interface Props {
  title: string;
  sections: string[];
  subSections?: { [key: string]: string[] };
  displayState?: 'DISPLAYED' | 'HIDDEN';
}

export const BurgerMenu: React.FC<Props> = ({
  title,
  sections,
  subSections = {},
  displayState = 'DISPLAYED',
}) => {
  const burgerButtonClassName = `fixed w-6 h-6  right-2.5 desktop:hidden transition-all delay-100 duration-300 ${
    displayState === 'HIDDEN' ? '-top-21.5' : 'top-2.5'
  }`;

  return (
    <Slide
      right
      customBurgerIcon={<BmIcon className="text-white" />}
      customCrossIcon={<Cross size={14} className="mt-3" />}
      burgerButtonClassName={burgerButtonClassName}
      burgerBarClassName="bg-white"
      menuClassName="bg-white p-4"
      crossButtonClassName="left-5"
      crossClassName="bg-greyDarkColored"
    >
      <span className="pb-4 font-bold text-center border-b border-solid border-greySoft outline-none">
        {title}
      </span>
      {sections.map(section => (
        <BurgerMenuSection title={section} subSections={subSections[section]} key={section} />
      ))}
    </Slide>
  );
};

export default BurgerMenu;
