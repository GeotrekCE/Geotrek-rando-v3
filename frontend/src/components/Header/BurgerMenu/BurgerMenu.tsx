import { useRouter } from 'next/router';
// @ts-ignore Not official but useful to reduce bundle size
import Slide from 'react-burger-menu/lib/menus/slide';
import { MenuConfig, MenuItem } from 'modules/header/interface';
import { useIntl } from 'react-intl';
import NextLink from 'next/link';
import { routes } from 'services/routes';
import { getDefaultLanguage } from '../../../modules/header/utills';
import { BurgerMenuSection } from '../BurgerMenuSection/BurgerMenuSection';
import { BurgerMenu as BmIcon } from '../../Icons/BurgerMenu';
import { Cross } from '../../Icons/Cross';

interface Props {
  config: MenuConfig;
  menuItems?: MenuItem[];
  displayState?: 'DISPLAYED' | 'HIDDEN';
}

export const BurgerMenu: React.FC<Props> = ({ config, menuItems, displayState = 'DISPLAYED' }) => {
  const burgerButtonClassName = `fixed w-6 h-6  right-2.5 desktop:hidden transition-all delay-100 duration-300 ${
    displayState === 'HIDDEN' ? '-top-21.5' : 'top-2.5'
  }`;

  const intl = useIntl();
  const router = useRouter();
  const currentLanguage = router.locale ?? getDefaultLanguage();

  return (
    <Slide
      id="verticalMenu"
      right
      customBurgerIcon={<BmIcon className="text-white" />}
      customCrossIcon={<Cross size={24} />}
      burgerButtonClassName={burgerButtonClassName}
      burgerBarClassName="bg-white"
      menuClassName="bg-white p-4"
      // We use mt-2 because we can't easily override the default element style with tailwind (default is top: 8 and we would like top: 16)
      crossButtonClassName="left-4 mt-2"
      crossClassName="bg-greyDarkColored"
    >
      <span
        id="verticalMenu_title"
        className="pb-4 font-bold text-center border-b border-solid border-greySoft outline-none"
      >
        {intl.formatMessage({ id: 'header.menu' })}
      </span>
      {menuItems && (
        <BurgerMenuSection title={intl.formatMessage({ id: 'header.seeMore' })} items={menuItems} />
      )}
      {config.shouldDisplayFavorite && (
        <BurgerMenuSection title={intl.formatMessage({ id: 'header.favorites' })} />
      )}
      <BurgerMenuSection
        title={intl.formatMessage({ id: 'header.language' })}
        languages={config.supportedLanguages}
      />
      <NextLink href={routes.SEARCH} passHref locale={currentLanguage} key={routes.SEARCH}>
        <a className="flex items-center pt-4 pb-4 font-bold outline-none cursor-pointer border-b pb-2 border-solid border-greySoft">
          {intl.formatMessage({ id: 'header.goToSearch' })}
        </a>
      </NextLink>
      <NextLink href={routes.OFFLINE} passHref locale={currentLanguage} key={routes.OFFLINE}>
        <a className="flex items-center pt-4 pb-4 font-bold outline-none cursor-pointer border-b pb-2 border-solid border-greySoft">
          {intl.formatMessage({ id: 'header.offline' })}
        </a>
      </NextLink>
    </Slide>
  );
};
