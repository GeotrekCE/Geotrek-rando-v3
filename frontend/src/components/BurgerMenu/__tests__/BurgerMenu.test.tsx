import { render } from 'services/testing/reactTestingLibraryWrapper';

import BurgerMenu from '../';

test('AAU, I can see a BurgerMenu', () => {
  const component = render(<BurgerMenu title="Test" sections={['Section 1', 'Section 2']} />);
  expect(component).toMatchSnapshot();
});
