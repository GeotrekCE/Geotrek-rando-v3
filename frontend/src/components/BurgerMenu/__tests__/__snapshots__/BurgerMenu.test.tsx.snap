// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AAU, I can see a BurgerMenu 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="bm-overlay"
          style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
        />
        <div>
          <div
            class="bm-burger-button fixed w-5 h-3 top-4 right-3"
            style="z-index: 1000;"
          >
            <button
              id="react-burger-menu-btn"
              style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
            >
              Open Menu
            </button>
            <span>
              <span
                class="bm-burger-bars bg-white"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
              />
              <span
                class="bm-burger-bars bg-white"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
              />
              <span
                class="bm-burger-bars bg-white"
                style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
              />
            </span>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="bm-menu-wrap"
          id=""
          style="position: fixed; right: 0px; z-index: 1100; width: 300px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
        >
          <div
            class="bm-menu bg-white"
            style="height: 100%; box-sizing: border-box; overflow: auto;"
          >
            <nav
              class="bm-item-list"
              style="height: 100%;"
            />
          </div>
          <div>
            <div
              class="bm-cross-button left-3"
              style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
            >
              <button
                id="react-burger-cross-btn"
                style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
                tabindex="-1"
              >
                Close Menu
              </button>
              <span
                style="position: absolute; top: 6px; right: 14px;"
              >
                <span
                  class="bm-cross bg-greyDarkColored"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
                />
                <span
                  class="bm-cross bg-greyDarkColored"
                  style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="bm-overlay"
        style="position: fixed; z-index: 1000; width: 100%; height: 100%; background: rgba(0, 0, 0, 0.3); opacity: 0; transform: translate3d(100%, 0, 0); transition: opacity 0.3s, transform 0s 0.3s;"
      />
      <div>
        <div
          class="bm-burger-button fixed w-5 h-3 top-4 right-3"
          style="z-index: 1000;"
        >
          <button
            id="react-burger-menu-btn"
            style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
          >
            Open Menu
          </button>
          <span>
            <span
              class="bm-burger-bars bg-white"
              style="position: absolute; height: 20%; left: 0px; right: 0px; top: 0%; opacity: 1;"
            />
            <span
              class="bm-burger-bars bg-white"
              style="position: absolute; height: 20%; left: 0px; right: 0px; top: 40%; opacity: 1;"
            />
            <span
              class="bm-burger-bars bg-white"
              style="position: absolute; height: 20%; left: 0px; right: 0px; top: 80%; opacity: 1;"
            />
          </span>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="bm-menu-wrap"
        id=""
        style="position: fixed; right: 0px; z-index: 1100; width: 300px; height: 100%; transform: translate3d(100%, 0, 0); transition: all 0.5s;"
      >
        <div
          class="bm-menu bg-white"
          style="height: 100%; box-sizing: border-box; overflow: auto;"
        >
          <nav
            class="bm-item-list"
            style="height: 100%;"
          />
        </div>
        <div>
          <div
            class="bm-cross-button left-3"
            style="position: absolute; width: 24px; height: 24px; right: 8px; top: 8px;"
          >
            <button
              id="react-burger-cross-btn"
              style="position: absolute; left: 0px; top: 0px; z-index: 1; width: 100%; height: 100%; margin: 0px; padding: 0px; font-size: 0px; background: transparent; cursor: pointer;"
              tabindex="-1"
            >
              Close Menu
            </button>
            <span
              style="position: absolute; top: 6px; right: 14px;"
            >
              <span
                class="bm-cross bg-greyDarkColored"
                style="position: absolute; width: 3px; height: 14px; transform: rotate(45deg);"
              />
              <span
                class="bm-cross bg-greyDarkColored"
                style="position: absolute; width: 3px; height: 14px; transform: rotate(-45deg);"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
